tests/framework/impl.go

```diff
@@ -0,0 +1,25 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+// Implementation defines the interface required by the actual merge code to be used in Kubernetes and this test framework
+type Implementation interface {
+	// Apply returns the new post merge object and errors (including conflicts if any occur)
+	Apply(live, config YAMLObject, workflow string, force bool) (YAMLObject, error)
```

Comment:
Somehow we need to pass the schema to the implementation as well; we could construct it knowing the schema and pass `value.Value`s for the objects, or we could pass typed.TypedValue for the objects. Thoughts?

Reply:
Yeah, I thought about this too but wanted to build what we outlined in the doc first.

I tried to think about ways that might suite you. As this was what we outlined in the doc I thought setting the spec on the implementation when initializing would be best.
I have to look at the types you introduced and the merge code to have more of an opinion though.

Reply:
That's fine for the moment at least. We can add it into the implementation.